Spigot-API-Patches/0231-Add-PlayerShearBlockEvent.patch

```diff
@@ -0,0 +1,123 @@
+
+
+diff --git a/src/main/java/io/papermc/paper/event/block/PlayerShearBlockEvent.java b/src/main/java/io/papermc/paper/event/block/PlayerShearBlockEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..7d945de6394cf73dc47c715eb865477fde62d1fa
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/block/PlayerShearBlockEvent.java
+@@ -0,0 +1,111 @@
++package io.papermc.paper.event.block;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.EquipmentSlot;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.List;
++
++/**
++ * Called when a player uses sheers on a block.
++ * <p>
++ * This event is <b>not</b> called when breaking blocks with shears but instead only when a
++ * player uses the sheer item on a block to garner drops from said block and/or change its state.
++ * <p>
++ * Examples include shearing a pumpkin to turn it into a carved pumpkin or shearing a beehive to get honeycomb.
++ */
++public class PlayerShearBlockEvent extends PlayerEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled = false;
++    private final Block block;
++    private final ItemStack item;
++    private final EquipmentSlot hand;
++    private final List<ItemStack> drops;
++
++    public PlayerShearBlockEvent(@NotNull Player who, @NotNull Block block, @NotNull ItemStack item, @NotNull EquipmentSlot hand, @Nullable List<ItemStack> drops) {
++        super(who);
++        this.block = block;
++        this.item = item;
++        this.hand = hand;
++        this.drops = drops;
++    }
++
++    /**
++     * Gets the block being sheared in this event.
++     *
++     * @return The {@link Block} which block is being sheared in this event.
++     */
++    @NotNull
++    public Block getBlock() {
++        return block;
++    }
++
++    /**
++     * Gets the item used to shear the block.
++     *
++     * @return The {@link ItemStack} of the shears.
++     */
++    @NotNull
++    public ItemStack getItem() {
++        return item;
++    }
++
++    /**
++     * Gets the hand used to shear the block.
++     *
++     * @return Either {@link EquipmentSlot#HAND} OR {@link EquipmentSlot#OFF_HAND}.
++     */
++    @NotNull
++    public EquipmentSlot getHand() {
++        return hand;
++    }
++
++    /**
++     * Gets the resulting drops of this event.
++     *
++     * This may return null in the future if Mojang adds a use for shears that do not drop items.
++     *
++     * @return A {@link List list} of {@link ItemStack items} that will be dropped as result of this event.
++     */
++    @Nullable
++    public List<ItemStack> getDrops() {
++        return drops;
```

Comment:
This shouldn't be nullable, ex; making blocks which don't drop items, drop items

Reply:
No idea why I made this nullable as it's never null 0_o